### PR TITLE
Make the use of string literal type hints compatible with PEP 484

### PR DESCRIPTION
**Pull request by meatballs**
[Because:
- Not doing so causes F821 linting errors

https://peps.python.org/pep-0484/#forward-references
https://www.flake8rules.com/rules/F821.html

**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [x] Other

**Required reviews:** 
- [x] 1
- [ ] 2
- [ ] 3

**What this does:**
Removes all but 5 F821 linting errors. Those are genuine undefined name errors rather than type hint issues.

**Notes for reviewers:**
- The relevant section of the PEP484 doc is where it shows an example from Django of using module level imports to avoid circular import problems. I've used that technique several times in this PR.
- To see the remaining F821 issues using [ruff](https://github.com/astral-sh/ruff):
```bash
>>> ruff check --select F821 nucypher
nucypher/blockchain/eth/interfaces.py:523:23: F821 Undefined name `JSONRPCStdoutEmitter`
nucypher/cli/actions/select.py:98:22: F821 Undefined name `Staker`
nucypher/cli/config.py:56:23: F821 Undefined name `JSONRPCStdoutEmitter`
nucypher/cli/painting/deployment.py:119:72: F821 Undefined name `DispatcherDeployer`
nucypher/cli/painting/deployment.py:123:35: F821 Undefined name `DispatcherDeployer`
Found 5 errors.
```

### CODE REVIEW

**Comment by meatballs** [comment]
Ah. It's running more tests than I did locally. I'll fix it up...

File: nucypher/network/middleware.py [hunk 1]
```diff
@@ -29,7 +30,7 @@ def __init__(
         self,
         eth_provider_uri: Optional[str],
         registry: Optional["BaseContractRegistry"] = None,
```
Comment (derekpierre) [inline nucypher/network/middleware.py:32]: Looks like this is imported already.

```suggestion
        registry: Optional[BaseContractRegistry] = None,
```

Comment (meatballs) [reply nucypher/network/middleware.py:32]: Missed that one. Nice catch!

**Comment by meatballs** [comment]
Looks like it got rate limited on the last test run

**Review by derekpierre — APPROVED** [review]
🎸

File: nucypher/acumen/perception.py [hunk 2]
```diff
@@ -41,7 +42,7 @@ def empty(self):
 
 class FleetState:
     """
-    Fleet state as perceived by a local Ursula.
+    Fleet state as perceived by a local "lawful.Ursula".
```
[the remark this answers is not in the document]
Comment (meatballs) [reply nucypher/acumen/perception.py:45]: ah. more of a `%s/Ursula/lawful.Ursula/g` victim!

File: nucypher/acumen/perception.py [hunk 3]
```diff
@@ -57,7 +58,9 @@ class FleetState:
     """
 
     @classmethod
-    def new(cls, this_node: Optional['Ursula'] = None) -> 'FleetState':
+    def new(
+        cls, this_node: Optional["characters.lawful.Ursula"] = None
+    ) -> "FleetState":
```
[the remark this answers is not in the document]
Comment (meatballs) [reply nucypher/acumen/perception.py:63]: no, because it's defined within the same module

**Pull request merged**